src/test_l1_dsVideoPort.c

```diff
@@ -92,9 +92,114 @@ static int gTestID = 1;
 	}\
 }\
 
+#define MAX_RESOLUTIONS 10
+typedef struct {
+    int type;
+    int index;
+} ConnectedAudioPort_t;
+
+typedef struct {
+    uint16_t port_no;
+    uint16_t type_id;
+    char* name;
+    bool dtcp_supported;
+    bool hdcp_supported;
+	uint16_t numSupportedResolutions;
+    char* supportedResolutionNames[MAX_RESOLUTIONS];
+    ConnectedAudioPort_t connected_audio_ports;
+    char* defaultResolution; // Adjust based on expected resolution name length
+    int colorspaces;
+    int supported_color_depth_capabilities;
+    bool display_surrond;
+    int supported_tv_resolutions_capabilities;
+    int hdr_capabilities;
+    int hdcp_protocol_version;
+    int quantization_ranges;
+    int matrix_coefficients;
+} PortConfig_t;
+
 /* Global flags to support features */
 static bool extendedEnumsSupported=false; //Default to not supported
-
+static char* deviceType = NULL;
+static char* deviceName = NULL;
+static int number_of_ports = 0;
+static char* port_types_str = NULL;
+static int color_depth = 0;
+PortConfig_t *ports = NULL;
+
+void populate_device_config() {
```

Comment:
This function can be used across the test suites (L1, L2, etc.), so keep this function in `test_util.c` and the structure in `test_util.h`. Regarding the use of variable names, follow the general rules, such as using `gvariable_name` for global variables and naming maximum values like MAX_RESOLUTIONS for enum max values.